### PR TITLE
Update make_release_notes to support list of issues

### PR DESCRIPTION
**Pull request by morganchen12**
Renders
```
# 2019-08-20 -- v4.1.3
- [changed] Cleaned up the documents, unused macros, and folders. (#3490, #3537, #3556, #3498)
- [changed] Updated the header path to pod repo relative. (#3527)
- [fixed] Fix singleton functionality after a FirebaseApp is deleted and recreated. (#3411)
```
as
```
### {{messaging}}

* {{changed}} Cleaned up the documents, unused macros, and folders. ([#3490](//github.com/firebase/firebase-ios-sdk/issues/3490), [#3537](//github.com/firebase/firebase-ios-sdk/issues/3537), [#3556](//github.com/firebase/firebase-ios-sdk/issues/3556), [#3498](//github.com/firebase/firebase-ios-sdk/issues/3498))
* {{changed}} Updated the header path to pod repo relative. ([#3527](//github.com/firebase/firebase-ios-sdk/issues/3527))
* {{fixed}} Fix singleton functionality after a FirebaseApp is deleted and recreated. ([#3411](//github.com/firebase/firebase-ios-sdk/issues/3411))
```

### CODE REVIEW

File: scripts/make_release_notes.py [hunk 1]
```diff
@@ -201,13 +209,13 @@ def parse_url(self, m):
     return self.renderer.url(m.group(1))
 
   local_issue_link = re.compile(
-      r'\('           # opening paren
-      r'#(\d+)'       # hash and issue number
-      r'\)'           # closing paren
+      r'\(?'             # opening paren
```
Comment (wilhuff) [inline scripts/make_release_notes.py]: The opening paren shouldn't be optional, should it?

Basically: if you're willing to match `#issue-number` without the paren then we don't need it in the pattern at all.

Also, if you're willing to just match `#issue-number` then you can simplify this by just matching the `#(\d+)\b` without even matching the list. Treat them each independently as a single issue link to replace.

Comment (morganchen12) [reply scripts/make_release_notes.py]: You're right. Fixed a similar issue for the closing paren as well.

File: scripts/make_release_notes.py [hunk 2]
```diff
@@ -244,11 +252,11 @@ def read_changelog_section(filename, single_version=None):
     # Discard all lines until we see a heading that either has the version the
     # user asked for or any version.
     if single_version:
-      initial_heading = re.compile(r'^#.*%s' % re.escape(single_version))
+      initial_heading = re.compile(r'^# .*%s' % re.escape(single_version))
```
Comment (wilhuff) [inline scripts/make_release_notes.py]: Should these also take the form `#{1,6}`?

Comment (morganchen12) [reply scripts/make_release_notes.py]: Done.

**Pull request merged**